src/arrayMethodJoin.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joinedString = '';
+    let divider = separator;
+
+    if (divider === undefined) {
+      divider = ',';
+    }
+
+    if (this.length === 0) {
+      return joinedString;
+    }
```

Comment:
This block of code does nothing, if you pass the empty string the loop is not executed and you just return joinedString, which you set as empty string on line 8

Reply:
Agree, ty)